rust/private/rustc.bzl

```diff
@@ -185,26 +185,31 @@ def get_cc_user_link_flags(ctx):
     """
     return ctx.fragments.cpp.linkopts
 
-def get_linker_and_args(ctx, cc_toolchain, feature_configuration, rpaths):
+def get_linker_and_args(ctx, cc_toolchain, feature_configuration, rpaths, attrs = None):
```

Comment:
Probably just FYI at this moment, but I plan to refactor the rules to not access ctx.attr behind rustc_compile_action. Everything will have to be passed explicitly. This will make rustc_compile_action less context dependent, and more reusable for example from aspects.

Reply:
So actually this PR is a step in the right direction, and my further change would make attrs parameter non-optional (and very likely unpacked, so for example there will be a non optional 'deps' parameter)

Reply:
I wasn't sure if you wanted me to do that in this change or not but I added an additional commit to make it so.